### PR TITLE
RunOnFxThread annotation and Interceptor

### PR DESCRIPTION
**Pull request by ghazyami**
`@RunOnFxThread` annotation and interceptor to run methods on JavaFx's UI thread  `JavaFX Application Thread` inspired by [Gluon Ignite](https://github.com/gluonhq/ignite)

- Example usage
```java
VBox vBox = new VBox();

public void start(@Observes @PrimaryStage final Stage stage) {
    Scene scene = new Scene(vBox);
    stage.setScene(scene);
    stage.show();

    try (ExecutorService executorService = Executors.newVirtualThreadPerTaskExecutor()) {
        for (int i = 0; i < 10; i++) {
            executorService.submit(() -> showNode(createExpensiveNode()));
        }
    }
}

Node createExpensiveNode() {
    try {
        TimeUnit.SECONDS.sleep(3);
    } catch (Exception e) {
        e.printStackTrace();
    }
    return new Label(LocalTime.now().toString());
}

@RunOnFxThread
void showNode(Node node) {
    vBox.getChildren().add(node);
}
```

### CODE REVIEW

**Review by CodeSimcoe — COMMENTED** [review]
Added some minor feedback

**Comment by CodeSimcoe** [comment]
Also, bonus :

- update the documentation
- provide a test case

Anyway, this can be done later (and I can help !)

**Comment by ghazyami** [comment]
@CodeSimcoe  Thanks for the quick review
Planning to update docs and try to write a test as part of this PR, just need some more time.

**Comment by CodeSimcoe** [comment]
I have a concern about testing a graphic application.
We'll need the build process to have a working graphic environment.

**Comment by ghazyami** [comment]
I added basic docs page and a test case inspired from #6, have not tested on any CI environment

File: deployment/src/test/java/io/quarkiverse/fx/deployment/RunOnFxThreadTest.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+import io.quarkiverse.fx.PrimaryStage;
+import io.quarkiverse.fx.RunOnFxThread;
+import io.quarkus.test.QuarkusUnitTest;
+import javafx.application.Application;
+import javafx.stage.Stage;
+
+public class RunOnFxThreadTest {
+
+    private static final int LAUNCH_TIMEOUT_MS = 1_000;
+    public static final String JAVA_FX_APPLICATION_THREAD = "JavaFX Application Thread";
```
Comment (CodeSimcoe) [inline deployment/src/test/java/io/quarkiverse/fx/deployment/RunOnFxThreadTest.java:24]: Simplify with Platform.isFxApplicationThread()
And maybe we have no guaranty that the thread name could change is next versions

**Comment by CodeSimcoe** [comment]
@all-contributors add @ghazyami for code

**Pull request merged**